net/udp: Check UDP header's length attribute

tbursztyka opened the pull request:
Length shoudl be at least of UDP header size.

Reported-by: Ruslan Mstoi <ruslan.mstoi@intel.com>

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/udp: Check UDP header's length attribute

Length should be at least of UDP header size but not bigger than actual
payload size.

Reported-by: Ruslan Mstoi <ruslan.mstoi@intel.com>

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

pfalcon commented:
> Length shoudl

Typo.

> Reported-by: Ruslan Mstoi ruslan.mstoi@intel.com

No bug ticket?

Commit:
tests/net: Fix IPv6 and UDP test as UDP got stricter on checking header

Now length of the UDP header is checked, and obviously these test were
avoiding setting it properly.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

rmstoi commented on subsys/net/ip/udp.c:
```diff
@@ -147,13 +147,23 @@ struct net_udp_hdr *net_udp_input(struct net_pkt *pkt,
 	}
 
+	length = ntohs(udp_hdr->len);
+	if (length < sizeof(struct net_udp_hdr) ||
```
Could it be that the first check (length < ) is redundant? The second check verifies that the length is correct, hence it should work by itself?

tbursztyka replied:
Indeed, moreover that net_pkt_get_data() can be successful if only payload length is at least of sizeof(struct net_udp_hdr). So checking if length is equal to payload length is indeed sufficient then.
